### PR TITLE
doc: idx sdk readme

### PR DESCRIPTION
**Pull request by shuowu**

### CODE REVIEW

File: docs/idx.md [hunk 1]
```diff
@@ -0,0 +1,307 @@
...
+
+### Concepts
+
+#### Approaches
+
+You can work with these methods with `Up-Front` and `On-Demand` approaches, normally a mix of both approaches will be needed when user inputs are required in the middle of the flow (e.g. multiple factors auth). Below are the general explaination of these two approaches, more code examples will be provided with the specific methods.
+
+##### Up-Front approach
+
+You can provide a param bag based on your app's policy configuration and user inputs to drive the methods to communicate with [Okta's Identity Engine](https://developer.okta.com/docs/concepts/ie-intro/).
```
Comment (aarongranick-okta) [inline docs/idx.md:41]: How about "parameters" instead of "a param bag"

File: docs/idx.md [hunk 2]
```diff
@@ -0,0 +1,307 @@
...
+
+##### On-Demand approach
+
+You can provide minimum or even no param to call the methods, and the methods can drive you through the flow by indicating the `nextStep` and required `inputs`.
```
Comment (aarongranick-okta) [inline docs/idx.md:45]: here also, let's use the full word "parameters"

Comment (aarongranick-okta) [reply docs/idx.md:45]: instead of "can drive you through the flow" how about "can provide direction"

File: docs/idx.md [hunk 3]
```diff
@@ -0,0 +1,307 @@
...
+
+#### Response
+
+Most methods in this module resolve [IdxTransaction](https://github.com/okta/okta-auth-js/blob/master/lib/idx/types/index.ts) as the response. With [Okta's Identity Engine](https://developer.okta.com/docs/concepts/ie-intro/)'s dynamic nature, it's very important to understand how to follow the response to proceed to the next step.
+
+##### Flow related fields
+
+- `status`: This field indicates the status of the current flow.
+  - `IdxStatus.SUCCESS`: This status indicates the flow is end with successfully get tokens.
```
Comment (aarongranick-okta) [inline docs/idx.md:54]: "is end" -> "has ended"

File: docs/idx.md [hunk 4]
```diff
@@ -0,0 +1,307 @@
...
+  - `IdxStatus.PENDING`: This status indicates the flow is still in progress, check `nextStep` and `messages` (intermediate form errors) fields in the response to proceed.
+  - `IdxStatus.FAILURE`: This status indicates error happens in SDK level, check `error` field in the response for error handling.
+  - `IdxStatus.TERMINAL`: This status indicates the flow runs into a `terminal` state, check `messages` field to handle it.
```
Comment (aarongranick-okta) [inline docs/idx.md:57]: "runs into" -> "has run into"

File: samples/generated/express-direct-auth/config.js [hunk 5]
```diff
@@ -12,9 +12,9 @@ module.exports = function () {
     webServer: {
       port: 8080,
       oidc: {
-        clientId: CLIENT_ID || '',
-        clientSecret: CLIENT_SECRET || '',
-        issuer: ISSUER || '',
+        clientId: CLIENT_ID,
```
Comment (aarongranick-okta) [inline samples/generated/express-direct-auth/config.js:15]: are these changes intentional?

**Review by aarongranick-okta — APPROVED** [review]
a few suggestions

File: samples/generated/express-direct-auth/config.js (hunk 5 above)
Comment (shuowu) [reply samples/generated/express-direct-auth/config.js:15]: I think it's changed by Vijet, this PR just generates the new samples based on the template file.
